packages/adapter-cloudflare/index.js

```diff
@@ -16,11 +16,13 @@ export default function (options = {}) {
 			builder.rimraf(tmp);
 			builder.mkdirp(tmp);
 
+			await builder.generateFallback(path.join(dest, '404.html'));
```

Comment:
What happens if someone has a `404.html/+page.svelte`? Will it overwrite this fallback without issues? If yes, should we add a comment explaining it?
 ```suggestion
 			// generate 404.html first which can then be overridden by prerendering, if the user defined such a page
			await builder.generateFallback(path.join(dest, '404.html'));
```

Also: Do we need to take the base path into account here? Or can we be sure the deployment has control of the root, too, which makes this a valid global 404 fallback?

Reply:
We shouldn't take the base path into account, because Pages is only interested in a `404.html` file in the root

Reply:
> We shouldn't take the base path into account, because Pages is only interested in a `404.html` file in the root

It looks like cf pages will search subdirectories too (if the 404 is on a subdirectory, it will search that first. If it doesn’t find one there, it searches the parent directly, etc. until it gets to the root 404)
https://developers.cloudflare.com/pages/platform/serving-pages/#not-found-behavior

Reply:
ah, good point. Still, there's no reason not to have the same behaviour for requests higher than the basepath